### PR TITLE
[Storage] Introduce `azure-storage-extensions`

### PR DESCRIPTION
**Pull request by jalauzon-msft**
This change adds a new package, `azure-storage-extensions` (name subject to change), which contains the C-extension module for Storage crc64. 

Consequently, this change also changes Storage to use the new "AdvancedBuild" pipelines to enable building the C-extension module with cibuildwheel. Included in the change are some tweaks/fixes to the pipelines and infrastructure to support c-extension packages.

### CODE REVIEW

**Review by scbedd — APPROVED** [review]
Approving EngSys changes.

Possible improvement brought up by @jalauzon-msft , separate the `azure-storage-extension` to its own build / release.

File: sdk/storage/azure-storage-extensions/README.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+This package is not meant to be used standalone and is meant to accompany other Azure Storage Python SDK libraries. However it can be installed and used standalone as well.
+
+Install the Azure Storage Extensions library for Python with [pip](https://pypi.org/project/pip/):
+
+```bash
+pip install azure-storage-extensions
+```
+
+Please note that this package contains [C-Extension modules](https://docs.python.org/3/extending/extending.html) and therefore you must install the correct wheel for your
+particular platform. We destribute many pre-built wheels for a wide variety of platforms on PyPi and `pip` will attempt to install the wheel meant for your platform. If
```
Comment (vincenttran-msft) [inline sdk/storage/azure-storage-extensions/README.md:19]: nit: destribute -> distribute

File: sdk/storage/azure-storage-extensions/README.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+however, a wheel is not available for your platform, we also distrubute the source code on PyPi so `pip` will attempt to build the module at install time, assuming you have
```
Comment (vincenttran-msft) [inline sdk/storage/azure-storage-extensions/README.md:20]: nit: distrubute -> distribute

File: sdk/storage/azure-storage-extensions/setup.py [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    description=PACKAGE_PPRINT_NAME,
+    long_description=open('README.md', 'r').read(),
+    long_description_content_type='text/markdown',
+    license='MIT License',
+    author='Microsoft Corporation',
+    author_email='ascl@microsoft.com',
+    url='https://github.com/Azure/azure-sdk-for-python/tree/main/sdk/storage/azure-storage-extensions',
+    keywords="azure, azure sdk",
+    classifiers=[
+        "Development Status :: 5 - Production/Stable",
```
Comment (vincenttran-msft) [inline sdk/storage/azure-storage-extensions/setup.py:28]: Is this prod release? iirc I saw a beta tag but could be wrong

**Review by vincenttran-msft — APPROVED** [review]
🚢 LGTM! Just some small nits.

File: sdk/storage/azure-storage-extensions/setup.py (hunk 3 above)
Comment (jalauzon-msft) [reply sdk/storage/azure-storage-extensions/setup.py:28]: Yeah, its a beta. Fixed

**Pull request merged**